gwctl/pkg/policymanager/merger.go

```diff
@@ -149,10 +156,26 @@ func mergePolicy(parent, child Policy) (Policy, error) {
 	result.u.SetUnstructuredContent(resultUnstructured)
 	// Merging two policies means the targetRef no longer makes any sense since
 	// since they can be conflicting. So we unset the targetRef.
-	result.targetRef = common.ObjRef{}
+	result.targetRefs = mergeTargetRefs(parent.targetRefs, child.targetRefs)
```

Comment:
Say `parent` has `targetRefs: [a, b]` and `child` has `targetRefs: [a, c]`.  So the merged (effective) policy applies to `[a, b, c]`?

Sorry, I may need to educate myself on how policies to be merged are selected in the first place – or, more precisely, how the policies are assigned to a particular node in the resource model. It just triggered me the idea that, if you ask for the effective policy for `a`, it will return a merge of `parent` and `child`, along with the info that such effective policy also applies to `c`. However, that's not entire really true, because if you ask for the effective policy for `c`, it may return something identical to `child` only (not a necessarily the `parent+child` merge.)

_Concrete example_ – Assume `a == gateway-1`, `b == route-1`, and `c == route-2`, though with `route-1.parentRefs == [gateway-1]` and `route-2.parentRefs == [gateway-2]`. (Notice `route-2` is NOT parented by `gateway-1`.)

IOW:

```yaml
kind: Gateway
metadata:
  name: gateway-1
---
kind: Gateway
metadata:
  name: gateway-2
---
kind: HTTPRoute
metadata:
  name: route-1
spec:
  parentRefs:
  - kind: Gateway
    name: gateway-1
---
kind: HTTPRoute
metadata:
  name: route-2
spec:
  parentRefs:
  - kind: Gateway
    name: gateway-2
---
kind: Policy
metadata:
  name: parent
spec:
  targetRefs:
  - kind: Gateway
    name: gateway-1
  - kind: HTTPRoute
    name: route-1
---
kind: Policy
metadata:
  name: child
spec:
  targetRefs:
  - kind: Gateway
    name: gateway-1
  - kind: HTTPRoute
    name: route-2
```

Effective policies:

```
gateway-1 → parent + child (targetRefs: [gateway-1, route-1, route-2])
gateway 2 → ∅
route-1 → parent (targetRefs: [gateway-1, route-1])
route-2 → child (targetRefs: [gateway-1, route-2])
```

By reading the first line, if you trust `parent` applies to `route-2` (or that `child` applies to `route-1` analogously), you would be wrong.